### PR TITLE
HBASE-23804: Fix default master addr hostname in master registry

### PR DESCRIPTION
**Pull request by bharathv**
Master rpc server end point doesn't always bind to localhost when a
specific end point is configured. It looks up the hostname and
binds to the endpoint to which it resolves. MasterRegistry should
do the same when building the default server end point to talk to.

### CODE REVIEW

File: hbase-server/src/main/java/org/apache/hadoop/hbase/master/HMaster.java [hunk 1]
```diff
@@ -20,6 +20,7 @@
 import static org.apache.hadoop.hbase.HConstants.DEFAULT_HBASE_SPLIT_COORDINATED_BY_ZK;
 import static org.apache.hadoop.hbase.HConstants.HBASE_MASTER_LOGCLEANER_PLUGINS;
 import static org.apache.hadoop.hbase.HConstants.HBASE_SPLIT_WAL_COORDINATED_BY_ZK;
+import static org.apache.hadoop.hbase.HConstants.MASTER_HOSTNAME_KEY;
```
Comment (ndimiduk) [inline hbase-server/src/main/java/org/apache/hadoop/hbase/master/HMaster.java]: Unused import?

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 32s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  The patch appears to include 4 new or modified test files.  |
||| _ HBASE-18095/client-locate-meta-no-zookeeper Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 48s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   5m 41s |  HBASE-18095/client-locate-meta-no-zookeeper passed  |
| +1 :green_heart: |  compile  |   3m 14s |  HBASE-18095/client-locate-meta-no-zookeeper passed  |
| +1 :green_heart: |  checkstyle  |   2m 26s |  HBASE-18095/client-locate-meta-no-zookeeper passed  |
| +1 :green_heart: |  shadedjars  |   4m 45s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   4m 16s |  HBASE-18095/client-locate-meta-no-zookeeper passed  |
| +0 :ok: |  spotbugs  |   3m 59s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +1 :green_heart: |  findbugs  |  20m 30s |  HBASE-18095/client-locate-meta-no-zookeeper passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 22s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   5m  5s |  the patch passed  |
| +1 :green_heart: |  compile  |   3m  6s |  the patch passed  |
| +1 :green_heart: |  javac  |   3m  6s |  the patch passed  |
| -1 :x: |  checkstyle  |   2m 17s |  root: The patch generated 4 new + 408 unchanged - 1 fixed = 412 total (was 409)  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  xml  |   0m  2s |  The patch has no ill-formed XML file.  |
| +1 :green_heart: |  shadedjars  |   4m 45s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  hadoopcheck  |  16m 26s |  Patch does not cause any errors with Hadoop 2.8.5 2.9.2 or 3.1.2.  |
| +1 :green_heart: |  javadoc  |   4m 19s |  the patch passed  |
| +1 :green_heart: |  findbugs  |  20m 37s |  the patch passed  |
||| _ Other Tests _ |
| -1 :x: |  unit  |  80m 21s |  root in the patch failed.  |
| -1 :x: |  asflicense  |   3m  1s |  The patch generated 1 ASF License warnings.  |
|  |   | 190m 10s |   |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.hbase.replication.multiwal.TestReplicationKillMasterRSCompressedWithMultipleAsyncWAL |
|   | hadoop.hbase.replication.TestReplicationStatusBothNormalAndRecoveryLagging |
|   | hadoop.hbase.replication.TestReplicationKillMasterRSWithSeparateOldWALs |
|   | hadoop.hbase.replication.TestSerialReplication |
|   | hadoop.hbase.replication.multiwal.TestReplicationSyncUpToolWithMultipleAsyncWAL |
|   | hadoop.hbase.replication.TestSerialReplicationFailover |
|   | hadoop.hbase.replication.TestReplicationDisableInactivePeer |
|   | hadoop.hbase.replication.TestReplicationSyncUpTool |
|   | hadoop.hbase.replication.TestSyncReplicationMoreLogsInLocalCopyToRemote |
|   | hadoop.hbase.replication.TestReplicationDroppedTables |
|   | hadoop.hbase.replication.multiwal.TestReplicationEndpointWithMultipleAsyncWAL |
|   | hadoop.hbase.replication.TestReplicationMetricsforUI |
|   | hadoop.hbase.replication.TestReplicationStatusSourceStartedTargetStoppedWithRecovery |
|   | hadoop.hbase.replication.TestReplicationStatusSourceStartedTargetStoppedNewOp |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.5 Server=19.03.5 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1137/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/1137 |
| JIRA Issue | HBASE-23804 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile xml |
| uname | Linux 5a5079f92532 4.15.0-60-generic #67-Ubuntu SMP Thu Aug 22 16:55:30 UTC 2019 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-slave/workspace/Base-PreCommit-GitHub-PR_PR-1137/out/precommit/personality/provided.sh |
| git revision | HBASE-18095/client-locate-meta-no-zookeeper / d110c08dce |
| Default Java | 1.8.0_181 |
| checkstyle | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1137/1/artifact/out/diff-checkstyle-root.txt |
| unit | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1137/1/artifact/out/patch-unit-root.txt |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1137/1/testReport/ |
| asflicense | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1137/1/artifact/out/patch-asflicense-problems.txt |
| Max. process+thread count | 9777 (vs. ulimit of 10000) |
| modules | C: hbase-common hbase-client hbase-server . U: . |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1137/1/console |
| versions | git=2.11.0 maven=2018-06-17T18:33:14Z) findbugs=3.1.11 |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

File: hbase-server/src/main/java/org/apache/hadoop/hbase/master/HMaster.java (hunk 1 above)
Comment (bharathv) [reply hbase-server/src/main/java/org/apache/hadoop/hbase/master/HMaster.java]: Fixed.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m 27s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +0 :ok: |  shelldocs  |   0m  0s |  Shelldocs was not available.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  The patch appears to include 4 new or modified test files.  |
||| _ HBASE-18095/client-locate-meta-no-zookeeper Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 47s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   6m  7s |  HBASE-18095/client-locate-meta-no-zookeeper passed  |
| +1 :green_heart: |  compile  |   3m 35s |  HBASE-18095/client-locate-meta-no-zookeeper passed  |
| +1 :green_heart: |  checkstyle  |   2m 41s |  HBASE-18095/client-locate-meta-no-zookeeper passed  |
| +0 :ok: |  refguide  |   5m 51s |  branch has no errors when building the reference guide. See footer for rendered docs, which you should manually inspect.  |
| +1 :green_heart: |  shadedjars  |   5m 26s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   4m 22s |  HBASE-18095/client-locate-meta-no-zookeeper passed  |
| +0 :ok: |  spotbugs  |   4m 23s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +1 :green_heart: |  findbugs  |  22m 37s |  HBASE-18095/client-locate-meta-no-zookeeper passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 21s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   5m 37s |  the patch passed  |
| +1 :green_heart: |  compile  |   3m 23s |  the patch passed  |
| +1 :green_heart: |  javac  |   3m 23s |  the patch passed  |
| +1 :green_heart: |  checkstyle  |   2m 35s |  root: The patch generated 0 new + 408 unchanged - 1 fixed = 408 total (was 409)  |
| +1 :green_heart: |  shellcheck  |   0m  2s |  There were no new shellcheck issues.  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  xml  |   0m  2s |  The patch has no ill-formed XML file.  |
| +0 :ok: |  refguide  |   5m 26s |  patch has no errors when building the reference guide. See footer for rendered docs, which you should manually inspect.  |
| +1 :green_heart: |  shadedjars  |   5m  5s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  hadoopcheck  |  17m 36s |  Patch does not cause any errors with Hadoop 2.8.5 2.9.2 or 3.1.2.  |
| +1 :green_heart: |  javadoc  |   4m 42s |  the patch passed  |
| +1 :green_heart: |  findbugs  |  23m 32s |  the patch passed  |
||| _ Other Tests _ |
| -1 :x: |  unit  | 177m 59s |  root in the patch failed.  |
| -1 :x: |  asflicense  |   1m 48s |  The patch generated 1 ASF License warnings.  |
|  |   | 308m 44s |   |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.hbase.security.access.TestSnapshotScannerHDFSAclController |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.5 Server=19.03.5 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1137/2/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/1137 |
| JIRA Issue | HBASE-23804 |
| Optional Tests | dupname asflicense shellcheck shelldocs javac javadoc unit spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile xml refguide |
| uname | Linux fbbe3dbfc7c8 4.15.0-74-generic #84-Ubuntu SMP Thu Dec 19 08:06:28 UTC 2019 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-slave/workspace/Base-PreCommit-GitHub-PR_PR-1137/out/precommit/personality/provided.sh |
| git revision | HBASE-18095/client-locate-meta-no-zookeeper / d110c08dce |
| Default Java | 1.8.0_181 |
| refguide | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1137/2/artifact/out/branch-site/book.html |
| refguide | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1137/2/artifact/out/patch-site/book.html |
| unit | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1137/2/artifact/out/patch-unit-root.txt |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1137/2/testReport/ |
| asflicense | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1137/2/artifact/out/patch-asflicense-problems.txt |
| Max. process+thread count | 5533 (vs. ulimit of 10000) |
| modules | C: hbase-common hbase-client hbase-server . U: . |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1137/2/console |
| versions | git=2.11.0 maven=2018-06-17T18:33:14Z) shellcheck=0.7.0 findbugs=3.1.11 |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

File: hbase-server/src/main/java/org/apache/hadoop/hbase/master/HMaster.java (hunk 1 above)
Comment (bharathv) [reply hbase-server/src/main/java/org/apache/hadoop/hbase/master/HMaster.java]: Actually it is needed here, I moved it to HConstants, so have to explicitly import now.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 31s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  1s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  The patch appears to include 4 new or modified test files.  |
||| _ HBASE-18095/client-locate-meta-no-zookeeper Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 41s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   5m 26s |  HBASE-18095/client-locate-meta-no-zookeeper passed  |
| +1 :green_heart: |  compile  |   2m  8s |  HBASE-18095/client-locate-meta-no-zookeeper passed  |
| +1 :green_heart: |  checkstyle  |   2m 26s |  HBASE-18095/client-locate-meta-no-zookeeper passed  |
| +1 :green_heart: |  shadedjars  |   5m 10s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   1m 23s |  HBASE-18095/client-locate-meta-no-zookeeper passed  |
| +0 :ok: |  spotbugs  |   4m 27s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +1 :green_heart: |  findbugs  |   6m 27s |  HBASE-18095/client-locate-meta-no-zookeeper passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 16s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   5m 34s |  the patch passed  |
| +1 :green_heart: |  compile  |   1m 54s |  the patch passed  |
| +1 :green_heart: |  javac  |   1m 54s |  the patch passed  |
| +1 :green_heart: |  checkstyle  |   0m 25s |  The patch passed checkstyle in hbase-common  |
| +1 :green_heart: |  checkstyle  |   0m 29s |  The patch passed checkstyle in hbase-client  |
| +1 :green_heart: |  checkstyle  |   1m 22s |  hbase-server: The patch generated 0 new + 408 unchanged - 1 fixed = 408 total (was 409)  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  shadedjars  |   4m 38s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  hadoopcheck  |  16m  8s |  Patch does not cause any errors with Hadoop 2.8.5 2.9.2 or 3.1.2.  |
| +1 :green_heart: |  javadoc  |   1m 24s |  the patch passed  |
| +1 :green_heart: |  findbugs  |   6m 45s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   1m  9s |  hbase-common in the patch passed.  |
| +1 :green_heart: |  unit  |   0m 54s |  hbase-client in the patch passed.  |
| -1 :x: |  unit  |  76m 15s |  hbase-server in the patch failed.  |
| +1 :green_heart: |  asflicense  |   1m 40s |  The patch does not generate ASF License warnings.  |
|  |   | 150m 37s |   |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.hbase.replication.regionserver.TestReplicationSink |
|   | hadoop.hbase.replication.TestSyncReplicationActive |
|   | hadoop.hbase.replication.TestReplicationStatus |
|   | hadoop.hbase.replication.regionserver.TestRegionReplicaReplicationEndpoint |
|   | hadoop.hbase.replication.TestReplicationStatusBothNormalAndRecoveryLagging |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.5 Server=19.03.5 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1137/3/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/1137 |
| JIRA Issue | HBASE-23804 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile |
| uname | Linux e7d1a5ce91ee 4.15.0-58-generic #64-Ubuntu SMP Tue Aug 6 11:12:41 UTC 2019 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-slave/workspace/Base-PreCommit-GitHub-PR_PR-1137/out/precommit/personality/provided.sh |
| git revision | HBASE-18095/client-locate-meta-no-zookeeper / d110c08dce |
| Default Java | 1.8.0_181 |
| unit | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1137/3/artifact/out/patch-unit-hbase-server.txt |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1137/3/testReport/ |
| Max. process+thread count | 9841 (vs. ulimit of 10000) |
| modules | C: hbase-common hbase-client hbase-server U: . |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1137/3/console |
| versions | git=2.11.0 maven=2018-06-17T18:33:14Z) findbugs=3.1.11 |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

File: hbase-common/src/main/java/org/apache/hadoop/hbase/HConstants.java [hunk 2]
```diff
@@ -181,7 +181,11 @@
   /** Configuration key for the list of master host:ports **/
   public static final String MASTER_ADDRS_KEY = "hbase.masters";
 
-  public static final String MASTER_ADDRS_DEFAULT =  "localhost:" + DEFAULT_MASTER_PORT;
```
Comment (ndimiduk) [inline hbase-common/src/main/java/org/apache/hadoop/hbase/HConstants.java]: Head's up, this file is `InterfaceAudience.Public`, which means removals have to go through a full release deprecation cycle.

From the sub-section titled "Client API compatibility" in [the book](http://hbase.apache.org/book.html#hbase.versioning.post10):

> An API needs to be deprecated for a whole major version before we will change/remove it.
> * An example: An API was deprecated in 2.0.1 and will be marked for deletion in 4.0.0. On the other hand, an API deprecated in 2.0.0 can be removed in 3.0.0.

You'll need to retain the hold field -- annotate it as deprecated and add a java doc explaining that the other method could be used in its place. You can file a follow-on ticket with a fixVersion of 4.0.0 as a task to delete the field once we get there (and maybe leave a TODO referencing that ticket).

Sadly this constant is a field not a method, which means we have no way to logging a WARN for clients who make use of it.

File: hbase-common/src/main/java/org/apache/hadoop/hbase/HConstants.java [hunk 3]
```diff
@@ -181,7 +181,11 @@
...
+  // key to the config parameter of server hostname
+  // the specification of server hostname is optional. The hostname should be resolvable from
+  // both master and region server
+  public static final String RS_HOSTNAME_KEY = "hbase.regionserver.hostname";
```
Comment (ndimiduk) [inline hbase-common/src/main/java/org/apache/hadoop/hbase/HConstants.java]: Don't add these here. In [the book](http://hbase.apache.org/book.html#hbase_default_configurations), they're considered "expert" configuration params (and the `master` version appears to not be documented at all).

Instead, add them to the `DNS` class, from which they'll be used. Also, it's `InterfaceAudience.Private` so we have more maintenance/refactoring flexibility there. And don't lose their `InterfaceAudience.LimitedPrivate` annotations when you move them.

File: hbase-common/src/main/java/org/apache/hadoop/hbase/util/DNS.java [hunk 4]
```diff
@@ -66,4 +68,24 @@ public static String getDefaultHost(String strInterface, String nameserver)
...
     }
   }
+
+  /**
+   * Get the configured hostname for master/regionserver. Gets the default hostname if not specified
+   * in the configuration.
+   * @param conf Configuration to look up.
+   * @param isMaster True if master hostname needs to be looked up and false for regionserver.
+   */
+  public static String getHostname(Configuration conf, boolean isMaster)
```
[the remark this answers is not in the document]
Comment (bharathv) [reply hbase-common/src/main/java/org/apache/hadoop/hbase/util/DNS.java]: Think those patches have reasonable test coverage and I verified they pass for me locally. If we break something about binding to wildcard address, we should know it from the nightly failures I think (ex: testMiniClusterBindToWildcard).. (Not sure if I answered your question, did I get the question right?)

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   3m 43s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  The patch appears to include 4 new or modified test files.  |
||| _ HBASE-18095/client-locate-meta-no-zookeeper Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 34s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   5m 53s |  HBASE-18095/client-locate-meta-no-zookeeper passed  |
| +1 :green_heart: |  compile  |   1m 53s |  HBASE-18095/client-locate-meta-no-zookeeper passed  |
| +1 :green_heart: |  checkstyle  |   2m 14s |  HBASE-18095/client-locate-meta-no-zookeeper passed  |
| +1 :green_heart: |  shadedjars  |   5m  7s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   1m 19s |  HBASE-18095/client-locate-meta-no-zookeeper passed  |
| +0 :ok: |  spotbugs  |   4m 46s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +1 :green_heart: |  findbugs  |   6m 41s |  HBASE-18095/client-locate-meta-no-zookeeper passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 14s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   5m 35s |  the patch passed  |
| +1 :green_heart: |  compile  |   1m 48s |  the patch passed  |
| +1 :green_heart: |  javac  |   1m 48s |  the patch passed  |
| +1 :green_heart: |  checkstyle  |   0m 24s |  The patch passed checkstyle in hbase-common  |
| +1 :green_heart: |  checkstyle  |   0m 29s |  The patch passed checkstyle in hbase-client  |
| +1 :green_heart: |  checkstyle  |   1m 20s |  hbase-server: The patch generated 0 new + 408 unchanged - 1 fixed = 408 total (was 409)  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  shadedjars  |   5m  9s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  hadoopcheck  |  17m 45s |  Patch does not cause any errors with Hadoop 2.8.5 2.9.2 or 3.1.2.  |
| +1 :green_heart: |  javadoc  |   1m 21s |  the patch passed  |
| +1 :green_heart: |  findbugs  |   7m 50s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   1m 15s |  hbase-common in the patch passed.  |
| +1 :green_heart: |  unit  |   1m  2s |  hbase-client in the patch passed.  |
| -1 :x: |  unit  | 136m 46s |  hbase-server in the patch failed.  |
| +1 :green_heart: |  asflicense  |   1m 57s |  The patch does not generate ASF License warnings.  |
|  |   | 218m 16s |   |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.hbase.master.assignment.TestCloseRegionWhileRSCrash |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.5 Server=19.03.5 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1137/4/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/1137 |
| JIRA Issue | HBASE-23804 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile |
| uname | Linux 140e0b331b89 4.15.0-74-generic #84-Ubuntu SMP Thu Dec 19 08:06:28 UTC 2019 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-slave/workspace/Base-PreCommit-GitHub-PR_PR-1137/out/precommit/personality/provided.sh |
| git revision | HBASE-18095/client-locate-meta-no-zookeeper / d110c08dce |
| Default Java | 1.8.0_181 |
| unit | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1137/4/artifact/out/patch-unit-hbase-server.txt |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1137/4/testReport/ |
| Max. process+thread count | 6916 (vs. ulimit of 10000) |
| modules | C: hbase-common hbase-client hbase-server U: . |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1137/4/console |
| versions | git=2.11.0 maven=2018-06-17T18:33:14Z) findbugs=3.1.11 |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   3m  5s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  The patch appears to include 4 new or modified test files.  |
||| _ HBASE-18095/client-locate-meta-no-zookeeper Compile Tests _ |
| +0 :ok: |  mvndep  |   1m 24s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   8m 45s |  HBASE-18095/client-locate-meta-no-zookeeper passed  |
| +1 :green_heart: |  compile  |   2m 38s |  HBASE-18095/client-locate-meta-no-zookeeper passed  |
| +1 :green_heart: |  checkstyle  |   3m 15s |  HBASE-18095/client-locate-meta-no-zookeeper passed  |
| +1 :green_heart: |  shadedjars  |   6m 42s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   1m 55s |  HBASE-18095/client-locate-meta-no-zookeeper passed  |
| +0 :ok: |  spotbugs  |   6m 42s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +1 :green_heart: |  findbugs  |   9m 40s |  HBASE-18095/client-locate-meta-no-zookeeper passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 17s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   7m 57s |  the patch passed  |
| +1 :green_heart: |  compile  |   2m 40s |  the patch passed  |
| +1 :green_heart: |  javac  |   2m 40s |  the patch passed  |
| +1 :green_heart: |  checkstyle  |   0m 36s |  The patch passed checkstyle in hbase-common  |
| +1 :green_heart: |  checkstyle  |   0m 42s |  The patch passed checkstyle in hbase-client  |
| +1 :green_heart: |  checkstyle  |   1m 46s |  hbase-server: The patch generated 0 new + 408 unchanged - 1 fixed = 408 total (was 409)  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  shadedjars  |   7m  0s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  hadoopcheck  |  19m  3s |  Patch does not cause any errors with Hadoop 2.8.5 2.9.2 or 3.1.2.  |
| +1 :green_heart: |  javadoc  |   1m 19s |  the patch passed  |
| +1 :green_heart: |  findbugs  |   7m  8s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   1m  8s |  hbase-common in the patch passed.  |
| +1 :green_heart: |  unit  |   0m 55s |  hbase-client in the patch passed.  |
| -1 :x: |  unit  |  99m 59s |  hbase-server in the patch failed.  |
| +1 :green_heart: |  asflicense  |   1m 19s |  The patch does not generate ASF License warnings.  |
|  |   | 197m 15s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.5 Server=19.03.5 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1137/5/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/1137 |
| JIRA Issue | HBASE-23804 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile |
| uname | Linux 00581c7182bf 4.15.0-74-generic #84-Ubuntu SMP Thu Dec 19 08:06:28 UTC 2019 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-slave/workspace/Base-PreCommit-GitHub-PR_PR-1137/out/precommit/personality/provided.sh |
| git revision | HBASE-18095/client-locate-meta-no-zookeeper / d0af674e7d |
| Default Java | 1.8.0_181 |
| unit | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1137/5/artifact/out/patch-unit-hbase-server.txt |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1137/5/testReport/ |
| Max. process+thread count | 6112 (vs. ulimit of 10000) |
| modules | C: hbase-common hbase-client hbase-server U: . |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1137/5/console |
| versions | git=2.11.0 maven=2018-06-17T18:33:14Z) findbugs=3.1.11 |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

File: hbase-common/src/main/java/org/apache/hadoop/hbase/HConstants.java (hunk 2 above)
Comment (ndimiduk) [reply hbase-common/src/main/java/org/apache/hadoop/hbase/HConstants.java]: Ah okay. Let's avoid adding any fields to `HConstants` on the feature branch if we can.

Comment (bharathv) [reply hbase-common/src/main/java/org/apache/hadoop/hbase/HConstants.java]: Ack.

**Pull request merged**